flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/FunctionService.java

```diff
@@ -87,11 +108,22 @@ public static UserDefinedFunction createFunction(
 			new FunctionDescriptorValidator().validate(properties);
 		}
 
-		// instantiate
-		Object instance = generateInstance(
-				HierarchyDescriptorValidator.EMPTY_PREFIX,
-				properties,
-				classLoader);
+		Object instance;
+		switch (properties.getString(FunctionDescriptorValidator.FROM)) {
+			case FunctionDescriptorValidator.FROM_VALUE_CLASS:
+				// instantiate
+				instance = generateInstance(
+					HierarchyDescriptorValidator.EMPTY_PREFIX,
+					properties,
+					classLoader);
+				break;
+			case FunctionDescriptorValidator.FROM_VALUE_PYTHON:
+				String fullyQualifiedName = properties.getString(PythonFunctionValidator.FULLY_QUALIFIED_NAME);
+				instance = PythonFunctionUtils.getPythonFunction(fullyQualifiedName, config);
+				break;
+			default:
+				throw new ValidationException("Unsupported function descriptor: " + properties);
```

Comment:
There is no need to make the `properties` in the exception. Just `properties.getString(FunctionDescriptorValidator.FROM)` is enough? What do you think?